src/main/java/uk/ac/ebi/embl/api/validation/fixer/entry/DivisionFix.java

```diff
@@ -86,6 +82,39 @@ public ValidationResult check(Entry entry) throws ValidationEngineException {
         return result;
     }
     
+    private String getDivisionByTaxid(SourceFeature primarySF){
+        String division = "";
+        if(primarySF.getTaxId()!=null && getDivisionFromCache(primarySF.getTaxId().toString())!=null){
+            division = getDivisionFromCache(primarySF.getTaxId().toString());
+        }else {
+            division = new TaxonomyClientImpl().getTaxonByTaxid(primarySF.getTaxId()).getDivision();
+            saveDivisionCache(primarySF.getTaxId().toString(), division);
+        }
+        return division;
+    }
+    
+    private String getDivisionByScientificName(SourceFeature primarySF){
+        String division = "";
+        if(primarySF.getScientificName()!=null && getDivisionFromCache(primarySF.getScientificName())!=null){
+            division = getDivisionFromCache(primarySF.getScientificName());
```

Comment:
- Code simplification suggestion (with changes suggested below):
```
String division = getDivisionFromCache(primarySF.getScientificName());
if (empty(division)) {
    List<Taxon> taxonList;
    ....
```

Reply:
Code simplification suggestion is added